dask/dataframe/io/io.py

```diff
@@ -151,8 +152,14 @@ def from_array(x, chunksize=50000, columns=None, meta=None):
     return new_dd_object(dsk, name, meta, divisions)
 
 
+from_array = dataframe_io_dispatch.register_function(
+    "from_array",
+    docstring=from_array_pandas.__doc__,
+)
+
+
```

Comment:
(Just an thought -- not suggesting changes at this point) This pattern is popping up in several places. I wonder if instead of 

```python
def from_array_pandas(x, chunksize=50000, columns=None, meta=None):
    # This function used to just be called `from_array` but we've needed now need to append `_pandas`
    ...

from_array = dataframe_io_dispatch.register_function(
    "from_array",
    docstring=from_array_pandas.__doc__,
)
```

which is a bit verbose, we could do something like

```python
@dataframe_io_dispatch.register("pandas")
def from_array(x, chunksize=50000, columns=None, meta=None):
    ...
```

in the `dask/dask` codebase  and then other projects that are implementing their own backends would look similar

```python
# somwhere in the `rapids/cudf` repo

@dataframe_io_dispatch.register("cudf")
def from_array(x, chunksize=50000, columns=None, meta=None):
    ...
```

That would prevent the need to append `_pandas` to all the existing functions in `dask/dask`. 

Just a thought, I could be missing some reason as to why we can't do this

Reply:
I just made some significant changes to this PR, and the new version is doing something a lot closer to what you are suggesting here.